core/src/main/java/tc/oc/pgm/kits/KitMatchModule.java

```diff
@@ -172,4 +173,15 @@ public void checkItemTransfer(ItemTransferEvent event) {
       event.setCancelled(true);
     }
   }
+
+  @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+  public void checkInfiniteBlocks(BlockPlaceEvent event) {
+    if (ItemTags.INFINITE.get(event.getPlayer().getItemInHand()) != null) {
+      ItemStack blockPlacedItem = new ItemStack(event.getBlockPlaced().getType());
+      // infinite block contains -1 items, giving -1 items sets the amount back to -1
+      blockPlacedItem.setAmount(-1);
+      ItemTags.INFINITE.set(blockPlacedItem, true);
+      event.getPlayer().setItemInHand(blockPlacedItem);
```

Comment:
This is not properly retaining block metadata. If the material is wool, you will be losing the color of said wool.
You should not be creating new itemstacks, instead, update the existing stack in the player inventory by setting the amount back to what it was.